### DISCUSSION TITLE
Add index to Item

### DISCUSSION THREAD

**Discussion by exectails** [NoCode-NoLife/melia#178] [Ideas]
When I initially coded the inventory, I thought it would be nice to keep the inventory index out of the Item class, so it doesn't has to know anything about that stuff. While this still works, maybe it would be easier to just set the index on the item, so we don't have to jump through hoops to get items with their indices, like creating indexed dictionaries or tuples. Right now you can't get an item with its index in one object, I think that's not ideal.

**Comment by ovflowd** [comment]
You're talking about the `Item` model storing information about his correspondent inventory id? Like doing a Foreign Key. In relational terms that works, but better than only storing the ID of the Inventory related to a User, in my humility opinion (i don't know anything about the source actually), is better a property related to the Inventory Model instance from a specific User.

**Comment by exectails** [comment]
In TOS the index basically dictates the order in which the items appear in your inventory. The indices also use ranges to determine in which category the items should appear. For example, 1000\~2000 might be weapons, 2000\~3000 might be armors, etc.

The client regularly wants item data together with the indices though, and getting the index for an item which you need to send to the client is a little annoying right now, because you'd have to ask the inventory to find the item and get its index.

The index is otherwise irrelevant and we don't even store it. We only store an int for the sort order, to keep the order the same on load next time. The index is assigned when the item is added to the inventory iirc.

https://github.com/aura-project/melia/blob/24ed668af3d7792539b45b553736316ce887b506/src/Shared/Const/Items.cs#L35
